multiqc/modules/deeptools/plotEnrichment.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+
+""" MultiQC submodule to parse output from deepTools plotEnrichment """
+
+import logging
+import re
+from collections import OrderedDict
+
+from multiqc import config
+from multiqc.plots import bargraph
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+class plotEnrichmentMixin():
+    def parse_plotEnrichment(self):
+        """Find plotEnrichment output."""
+        self.deeptools_plotEnrichment = dict()
+        for f in self.find_log_files('deeptools/plotEnrichment'):
+            parsed_data = self.parsePlotEnrichment(f['f'], f['fn'])
+            for k, v in parsed_data.items():
+                if k in self.deeptools_plotEnrichment:
+                    log.warning("Replacing duplicate sample {}.".format(k))
+                self.deeptools_plotEnrichment[k] = v
+
+            if len(parsed_data) > 0:
+                self.add_data_source(f, section='plotEnrichment')
+
+        if len(self.deeptools_plotEnrichment) > 0:
+            dCounts = OrderedDict()
+            dPercents = OrderedDict()
+            for sample, v in self.deeptools_plotEnrichment.items():  # Key is sample
+                for category, v2 in v.items():  # Key is categ
+                    if category not in dCounts:
+                        dCounts[category] = dict()
+                        dPercents[category] = dict()
+                    dCounts[category][sample] = v2['count']
+                    dPercents[category][sample] = v2['percent']
+            config = {'data_labels': [
+                          {'name': 'Counts in features', 'ylab': 'Counts in feature'},
+                          {'name': 'Percents in features', 'ylab': 'Percent of reads in feature'}],
+                      'ylab': 'Counts in feature',
+                      'stacking': None,
```

Comment:
Cool! This is the only example of this feature being used that I can remember. However, I'm not sure that it can stay... It looks really cool in the demo data with two samples, but it won't scale well at all. Because of people like @FelixKrueger doing single-cell work / other examples, MultiQC reports are often generated with hundreds or thousands of samples and this has a habit of breaking everything. Having split bars where each bar is a sample is a prime example of this - bar plots break down at the best of times with large sample numbers and here we're effectively multiplying the number of samples by eight.

My suggestion would be to replace this with a line graph that has a categorical x axis. It's not ideal, but it would have one line per sample, which would scale well.

Reply:
Hmmm, true, we happen to not use that particular tool for our single-cell stuff. I'll see if I can get a useful line plot.